com.unity.render-pipelines.universal/Shaders/PostProcessing/EdgeAdaptiveSpatialUpsampling.shader

```diff
@@ -0,0 +1,59 @@
+Shader "Hidden/Universal Render Pipeline/Edge Adaptive Spatial Upsampling"
+{
+    HLSLINCLUDE
+        #pragma multi_compile _ _USE_DRAW_PROCEDURAL
+
+        #include "Packages/com.unity.render-pipelines.core/ShaderLibrary/Common.hlsl"
+        #include "Packages/com.unity.render-pipelines.core/ShaderLibrary/Filtering.hlsl"
+        #include "Packages/com.unity.render-pipelines.universal/ShaderLibrary/Core.hlsl"
+        #include "Packages/com.unity.render-pipelines.universal/Shaders/PostProcessing/Common.hlsl"
+
+        TEXTURE2D_X(_SourceTex);
+        float4 _SourceSize;
+
+        #define FSR_INPUT_TEXTURE _SourceTex
+        #define FSR_INPUT_SAMPLER sampler_LinearClamp
+
+        // Only enable 16-bit instructions when the underlying hardware supports them
+        // Note: There are known issues on DX11 drivers so we don't enable 16-bit mode when DX11 is detected
+        #if HAS_HALF && !defined(SHADER_API_D3D11)
```

Comment:
This is interesting, Looking at the code in common.hlsl it seems we should use REAL_IS_HALF and you should use `real` instead of `half` in any case that you want fp16. This should work ™️  but well, I know expectations are hard when it comes to drivers 🤣. Either way, I think this kind of check should not be done at URP level, URP should consume REAL_IS_HALF and if there's any kind of specific platform issue / driver issue the exception for DX11 should be handled at the definion of REAL_IS_HALF 
https://sourcegraph.com/github.com/Unity-Technologies/Graphics/-/blob/com.unity.render-pipelines.core/ShaderLibrary/Common.hlsl?L85